### ISSUE TITLE
Extra sample detected in merged MAF File

### ISSUE THREAD

**Issue by aditisk** [PoisonAlien/maftools#475]
Hello, I am new MAFtools user and I'm having trouble with loading my MAF files.

I am using "cat" to combine my MAF files into a single MAF. I have set the tumor_sample_barcode and I should have 14 samples. However, MAFtools is reading it as 15 samples with 1 sample having an "unknown" tumor_sample_barcode. 

Any thoughts on why this might be happening and how to fix this ? Thanks.

**Comment by ShixiangWang** [comment]
You can read your data with fread and check the column for sample. I think this is not a maftools issue

**Comment by aditisk** [comment]
@ShixiangWang I'll try that. Thanks for the suggestion.

For a different project I'm having a similar issue of an extra sample being detected. In the output of getSampleSummary() I see 21 samples but in the Oncoplot (attached here) the sample number is 22. Why is this happening ?

Thanks for your help.
![Rplot](https://user-images.githubusercontent.com/15878219/75466141-bad88400-5957-11ea-9e60-dcbfa6217641.png)

**Comment by aditisk** [comment]
I just realized that there are 21 columns in the Oncoplot which is correct. Are the % on the right side don't seem correct unless there is some rounding off ? Like for TP53, 45% is correct if we consider 22 samples but it should be 47% for 21 samples (which is the actual number of samples).

**Comment by PoisonAlien** [comment]
Hello,
`read.maf` can read multiple maf files. If you dont want to `cat`, you can just pass all your mafs to the function and it should just work fine. Or there is also a function `merge_mafs()` for merging several mafs into a single entity. 

For the second question, I guess its due to that `unknown` sample being counted. You can check this with `getSampleSummary()`.

**Comment by ShixiangWang** [comment]
I agree with @PoisonAlien. I counted several times for columns of oncoplot, 21 is right as shown by the text `21 of 22`. The plot tells you the gene mutations cover 21 samples but the total sample number 22 is used by percent calculation. This issue comes from data but not plot.

**Comment by aditisk** [comment]
Hello @ShixiangWang and @PoisonAlien. 

I think there is some confusion here. The “unknown” sample was getting detected for a different project. For that, I’ll try the functions mentioned by @PoisonAlien instead of cat to merge MAFs.

The plot above is from a different project where I don’t have the issue of an “unknown” sample being detected. In the summary I see 21 samples but the plot header says 22. I don’t think the data is the source of the error since I see 21 samples in summary (without any “unknown”). How can I fix this ? 

Thank you.

**Comment by PoisonAlien** [comment]
Can you try `getSampleSummary` and check the entries under `Tumor_Sample_Barcode` column? It should list all 22 sample names, including the extra culprit.

**Comment by aditisk** [comment]
![image](https://user-images.githubusercontent.com/15878219/75552957-71df0900-5a05-11ea-9bc1-3726ba36447f.png)

As you can see, there are only 21 samples here.

**Comment by PoisonAlien** [comment]
I see. I cant seem to guess the possible issue. Can you draw the plot with following arguments and see if it still shows 21 samples (columns)?

```r
oncoplot(maf = yourmaf, showTumorSampleBarcodes = TRUE, removeNonMutated = FALSE)
```

**Comment by aditisk** [comment]
![Rplot01](https://user-images.githubusercontent.com/15878219/75562959-7a403f80-5a17-11ea-863f-7c4a1d80e676.png)

It now shows 22 columns. Where is the "_Barcode" coming from since it is not in the summary ? 

Thanks for your help in getting this sorted out.

**Comment by PoisonAlien** [comment]
It's there in the maf. However it doesn't seem to have any non-synonymous variants hence its not showing up in the main maf slot. You will have to check in you maf file for more details. I am guessing its due to the merging of the multiple files.

**Comment by ShixiangWang** [comment]
I think the data is small. Could you provide here for checking？

**Comment by aditisk** [comment]
![Rplot02](https://user-images.githubusercontent.com/15878219/75610285-f309cf00-5add-11ea-8b4e-7d76461246f9.png)

Hi @PoisonAlien and @ShixiangWang. I finally was able to figure out what was causing the issue. When I used "cat" to merge by MAFs, I deleted all lines with # but each MAF file still had a header. As a result, I had 21 headers in the merged MAF file. Those repetitive header lines were considered as an extra sample and hence the plot was showing 22 samples instead of 21.

Thanks for your help.
